File: service/adapter/builders/RouterLogBuilder.js
class RouterLogBuilder {
  constructor() {
    this.reset();
  }

  reset() {
    this.log = {};
    this.raw = '';
  }

  withRaw(raw) {
    this.raw = raw;
    return this;
  }

  // Example parse implementation
  parse(raw) {
    this.withRaw(raw);
    const parts = String(this.raw).trim().split('|');
    this.log.timestamp = parts[0] || null;
    this.log.level = parts[1] || null;
    this.log.message = parts.slice(2).join('|') || null;
    return this;
  }

  build() {
    const result = this.log;
    this.reset();
    return result;
  }
}

module.exports = RouterLogBuilder;

File: dependency-mapper/dependency-mapper/src/main/java/com/example/mapper/adapter/builders/RouterLogBuilder.java
package com.example.mapper.adapter.builders;

import com.example.mapper.adapter.RouterLog;
import java.util.Arrays;

public class RouterLogBuilder {
    private RouterLog log = new RouterLog();

    public RouterLogBuilder parse(String raw) {
        String[] parts = raw.trim().split("\\|");
        if (parts.length > 0) log.setTimestamp(parts[0]);
        if (parts.length > 1) log.setLevel(parts[1]);
        if (parts.length > 2) log.setMessage(String.join("|", Arrays.copyOfRange(parts, 2, parts.length)));
        return this;
    }

    public RouterLog build() {
        RouterLog result = log;
        log = new RouterLog();
        return result;
    }
}
